user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/config/CacheConfiguration.java

```diff
@@ -0,0 +1,210 @@
+package de.tum.in.www1.artemis.usermanagement.config;
+
+import java.util.Collections;
+
+import javax.annotation.PreDestroy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.web.ServerProperties;
+import org.springframework.boot.info.BuildProperties;
+import org.springframework.boot.info.GitProperties;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.cloud.client.serviceregistry.Registration;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.hazelcast.config.*;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.spring.context.SpringManagedContext;
+
+import tech.jhipster.config.JHipsterProperties;
+import tech.jhipster.config.cache.PrefixedKeyGenerator;
+
+@Configuration
+@EnableCaching
+public class CacheConfiguration {
+
+    private final Logger log = LoggerFactory.getLogger(CacheConfiguration.class);
+
+    private GitProperties gitProperties;
+
+    private BuildProperties buildProperties;
+
+    private final ServerProperties serverProperties;
+
+    private final DiscoveryClient discoveryClient;
+
+    private Registration registration;
+
+    private final ApplicationContext applicationContext;
+
+    @Value("${spring.jpa.properties.hibernate.cache.hazelcast.instance_name}")
+    private String instanceName;
+
+    @Value("${spring.hazelcast.interface:}")
+    private String hazelcastInterface;
+
+    @Value("${spring.hazelcast.port:5701}")
+    private int hazelcastPort;
+
+    @Value("${spring.hazelcast.localInstances:true}")
+    private boolean hazelcastLocalInstances;
+
+    public CacheConfiguration(ServerProperties serverProperties, DiscoveryClient discoveryClient, ApplicationContext applicationContext) {
+        this.serverProperties = serverProperties;
+        this.discoveryClient = discoveryClient;
+        this.applicationContext = applicationContext;
+    }
+
+    @Autowired(required = false) // ok
```

Comment:
What does this comment mean? Please elaborate.

Reply:
Same with other comments as this one

Reply:
This file was copied from Artemis. I couldn't reuse it because additional dependency to QuizScheduleService had to be added. I can just remove those comments, I suppose it was added to tell that it's okay to not require the autowiring